# Урок 10. Дисперсионный анализ.

1. Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов:  
Футболисты: 173, 175, 180, 178, 177, 185, 183, 182.  
Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180.  
Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [9]:
import numpy as np
import pandas as pd
from scipy import stats

In [10]:
football = ([173, 175, 180, 178, 177, 185, 183, 182])
hockey = ([177, 179, 180, 188, 177, 172, 171, 184, 180])
weightlifters = ([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

Нулевая гипотеза, что средний рост не зависит от представленного вида спорта.  
$ H0:\mu_1=\mu_2=\mu_3 $  
Альтернативная гипотеза, что средний рост спортсмена зависит от представленного вида спорта.  
$ H1:\mu_1\neq\mu_2\neq\mu_3 $  
Для решения возьмём встроенный метод однофакторного дисперсионного анализа из библиотеки scipy.

In [11]:
stats.f_oneway(football, hockey, weightlifters)

F_onewayResult(statistic=5.500053450812596, pvalue=0.010482206918698694)

Ответ:  
p-value = 0.01048,  
если уровень статистической значимости $ \alpha=0.05 $ то отвергаем нулевую теорию.  
То есть средний рост спортсмена зависит от представленного вида спорта.

2. В одной группе из 100 больных наблюдалось улучшение у 75, а в другой из 100 больных среди 69. Оценить с помощью доверительного интервала разность долей больных, у которых наблюдались улучшения.

Интервал разности долей находим по формуле  
$ \Delta\pm z_{a/2}*S_\Delta $, где  
$ \Delta $ - разность долей (df)  
$ S_\Delta $ - стандартная ошибко разности долей

In [26]:
n1 = 75  # больные из первой группы у которых наблюдалось улучшение
m1 = 100 # всего больных в первой группе
n2 = 69  # больные из второй группы у которых наблюдалось улучшение
m2 = 100 # всего больных во второй группе
z = 1.96 # находим по таблице

In [27]:
p1 = n1/m1
p1

0.75

In [28]:
p2 = n2/m2
p2

0.69

In [29]:
df = p1 - p2
df

0.06000000000000005

Найдём общую долю Pg

In [30]:
Pg = (n1 + n2)/(m1 + m2)
Pg

0.72

Найдём стандартную ошибку разности долей Sd

In [31]:
Sd = np.sqrt(Pg*(1-Pg)*(1/n1+1/n2))
Sd

0.07489819177012742

Вычислим интервал  
$ 0.06\pm1.96*0.075 $

In [32]:
df + z * Sd

0.2068004558694498

In [33]:
df - z * Sd

-0.0868004558694497

Ответ:  
[-0.087; 0.207]

3. Какова вероятность, что в наудачу выбранном двузначном числе цифры одинаковые?

Двузначных чисел 90 (10 - 99)  
Одинаковые цифры в 9 числах (11, 22, 33, 44, 55, 66, 77, 88, 99)

In [34]:
res = 9/90
res

0.1

Ответ:  
вероятность 0.1